backend/gn_plugin_depobio/demarches_simplifiees/file.py

```diff
@@ -0,0 +1,107 @@
+import dataclasses
+import datetime
+from enum import Enum
+from typing import Optional
+from geonature.utils.config import config
+
+configuration_depobio = config["PLUGIN_DEPOBIO"]
+
+
+class State(str, Enum):
+    EN_CONSTRUCTION = "en_construction"
+    ACCEPTE = "accepte"
+    EN_INSTRUCTION = "en_instruction"
+    REFUSE = "refuse"
+    SANS_SUITE = "sans_suite"
+
+
+@dataclasses.dataclass()
+class Field:
+    id: str
+    champDescriptorId: str
+    _typename: str
+    label: str
+    stringValue: str
+    prefilled: bool
+    columns: list
+    values: str = None
+    selected: Optional[bool] = None
+    date: Optional[datetime.date] = None
+
+
+@dataclasses.dataclass()
+class Fields:
+    fields: list[Field]
+
+    def __init__(self, fields: list[Field]):
+        self.fields = fields
+        self.__fields_as_dict = {field.id: field for field in fields}
+        self.configuration = configuration_depobio["DEMARCHES_SIMPLIFIEES"]
+
+    def get_libelle(self):
+        try:
+            return self.__fields_as_dict[self.configuration["LIBELLE_ID"]]
+        except KeyError as err:
+            raise ValueError(
+                f"The value supplied for the configuration of libelle"
+                f" ({self.configuration['LIBELLE_ID']}) appears to be wrong. Check your configs"
+            ) from err
+
+    def get_description(self):
+        try:
+            return self.__fields_as_dict[self.configuration["DESCRIPTION_ID"]]
+        except KeyError as err:
+            raise ValueError(
+                f"The value supplied for the configuration of description"
+                f" ({self.configuration['DESCRIPTION_ID']}) appears to be wrong. Check your configs"
+            ) from err
+
+    def get_date_fin(self):
+        try:
+            return self.__fields_as_dict[self.configuration["DATE_FIN_ID"]]
+        except KeyError as err:
+            raise ValueError(
+                f"The value supplied for the configuration of date_fin"
+                f" ({self.configuration['DATE_FIN_ID']}) appears to be wrong. Check your configs"
+            ) from err
+
+    @classmethod
+    def champs_from_list(cls, fields: list[dict]) -> "Fields":
+        fields_result = []
+        for champ in fields:
+            # We make copies so popping doesn't affect the original list
+            champ_copy = champ.copy()
+            # double underscore mess with python name mangling
+            champ_copy["_typename"] = champ_copy.pop("__typename")
+            fields_result.append(champ_copy)
+        return cls([Field(**champ_copy) for champ_copy in fields_result])
```

Comment:
```suggestion
    @staticmethod
    def champs_from_list(fields: list[dict]) -> "Fields":
        fields_result = []
        for champ in fields:
            # We make copies so popping doesn't affect the original list
            champ_copy = champ.copy()
            # double underscore mess with python name mangling
            champ_copy["_typename"] = champ_copy.pop("__typename")
            fields_result.append(champ_copy)
        return Fields([Field(**champ_copy) for champ_copy in fields_result])
```

Utilisation d'une méthode statique plus adaptée ici, car :
- Pas d'utilisation d'attributs de la classe
- Pas d'héritage de classe, ni actuellement ni à prévoir

Reply:
Alors, oui on peut utiliser une méthode statique, ça marcherait très bien. Toutefois, la création d'instance de la classe (factory) est un peu le cas d'usage typique de la méthode de classe. 
https://realpython.com/instance-class-and-static-methods-demystified/
https://www.geeksforgeeks.org/python/class-method-vs-static-method-python/
https://leandeep.com/static-method-vs-class-method-en-programmation-orient%C3%A9e-objet/

Le fait qu'on manipule la classe (en créant des instance) donne plutôt envie d'utiliser une classmethod que static